src/Rfc6455Client.php

```diff
@@ -401,7 +401,7 @@ private function onControlFrame(int $opcode, string $data): void
                 break;
 
             case Opcode::PING:
-                $this->write($data, Opcode::PONG)->ignore();
+                $this->write($data, Opcode::PONG);
```

Comment:
This is a behavioral change, as the parser will now block until the pong frame can be written.

Reply:
Would it make sense to make a writeAsync() function with `async(fn() => $this->write($data, $opcode, $rsv, $isFinal));` for all the `->ignore()` cases here?

Reply:
No, it would be better to use `AsyncWriter` from byte-stream for all the writes, then await them as needed. Using a new fiber for each pong frame is very suboptimal. I'm trying to decide if blocking the parser matters that much though.

Reply:
Also, thinking along the same line - would it make sense to make `sendAsync` and `sendBinaryAsync` methods to give this option on the public interface?

Reply:
Somewhat related: I meant to point out this subsequent fix in http-server so you were aware: https://github.com/amphp/http-server/commit/3f7d0f484fbef4d245493002838ade138b14d00c. The future return was to avoid blocking – using a blocking write and returning a completed future doesn't accomplish anything.

> would it make sense to make sendAsync and sendBinaryAsync methods to give this option on the public interface

I was considering that, since websocket-server will want to use an async send for broadcasting. I wonder if send and sendBinary should continue returning futures or if there should be separate methods.

Reply:
Alright I think I understand futures a bit more - in that case it makes more sense to provide a `send(): Future` and then people should `send()->await()` if they want it to block.  I think 88e668e and 7d764f0 should be implementing this pattern, although tests are failing even with latest v2 branch of byte-stream (possibly related: https://github.com/amphp/byte-stream/blob/v2/src/AsyncWriter.php#L106 should be `return Future::error` instead of `Future::error`?)

Reply:
>  in that case it makes more sense to provide a send(): Future and then people should send()->await() if they want it to block

Yes, that is what I was considering, but this is apart from the `write` API on sockets, so might not be the correct choice. I'm going to have to think about it some more. I can cherry pick the prior commits if need be.

> (possibly related: https://github.com/amphp/byte-stream/blob/v2/src/AsyncWriter.php#L106 should be return Future::error instead of Future::error?)

Oops, yes, of course. Thanks for pointing that out. Feel free to PR if you'd like.